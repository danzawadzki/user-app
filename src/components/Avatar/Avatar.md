React component example:

Default avatar.

```jsx
<Avatar
	name="Random User"
	avatar={{
		small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
		regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
		large: 'https://randomuser.me/api/portraits/men/65.jpg'
	}}
/>
```

Smaller avatar.

```jsx
<Avatar
	name="Random User"
	isSmaller={true}
	avatar={{
		small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
		regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
		large: 'https://randomuser.me/api/portraits/men/65.jpg'
	}}
/>
```

Avatar with out passed img src.

```jsx
<Avatar name="Random User" />
```
