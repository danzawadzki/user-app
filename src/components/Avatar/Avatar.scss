//TODO: Line overwrites rem to be multiple of 10px, not 16px. I keep it in the component styles, as I didn't know how to pass global styles to the react-styleguidist.
html {
	font-size: 62.5%;
}

.Avatar {
	background-image: url('../../assets/img/user_placeholder.png');
	background-size: cover;
	border-radius: 50%;
	width: 7rem;
	height: 7rem;

	&__image {
		border-radius: 50%;
		width: 7rem;
		height: 7rem;

		&--smaller {
			width: 4rem;
			height: 4rem;
		}
	}

	&--smaller {
		width: 4rem;
		height: 4rem;
	}
}
