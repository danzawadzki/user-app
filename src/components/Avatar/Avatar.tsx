import * as React from 'react';
import './Avatar.scss';

export interface IAvatarSrc {
	/** Optional image url for small screen devices (screen < 600px) */
	small?: string;
	/** Required image ulr for medium screen devices (600px =< dev <= 1600px) or all else if not provided. */
	regular?: string;
	/** Optional image url for large screen devices (1900px < screen) */
	large?: string;
}

export interface IAvatar {
	/** User name, who's avatar is displayed for alt attr. */
	name: string;
	/** Flag to display smaller avatar. */
	isSmaller?: boolean;
	/** Object containing avatar urls. */
	avatar?: IAvatarSrc | null;
}

/**
 * Avatar component.
 *
 * @version 1.0.0
 */
const Avatar = ({ name, isSmaller = false, avatar }: IAvatar) => {
	// Creating srcSet
	let srcSet = '';
	if (avatar) {
		if (avatar.small) {
			srcSet = `${avatar.small} 480w, `;
		}
		if (avatar.regular) {
			srcSet += `${avatar.regular} 960w, `;
		}
		if (avatar.regular) {
			srcSet += `${avatar.large} 1280w`;
		}
	}

	return (
		<div className={`Avatar ${isSmaller ? 'Avatar--smaller' : ''}`}>
			{srcSet && (
				<img
					className={`Avatar__image ${
						isSmaller ? 'Avatar__image--smaller' : ''
					}`}
					srcSet={srcSet && srcSet}
					alt={`${name} avatar`}
				/>
			)}
		</div>
	);
};

export default Avatar;
