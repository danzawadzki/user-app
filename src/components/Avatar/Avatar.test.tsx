import React from 'react';
import { shallow } from 'enzyme';
import Avatar from './Avatar';

describe('Avatar component: ', () => {
	//Shallow rendered component
	const name = 'Joe Black';
	const avatars = {
		small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
		regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
		large: 'https://randomuser.me/api/portraits/men/65.jpg'
	};
	const component = shallow(<Avatar name={name} avatar={avatars} />);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly with all props', () => {
		expect(component.find('.Avatar')).toBeDefined();
		expect(component.find('.Avatar')).not.toBeFalsy();
		expect(component.find('.Avatar__image').exists()).toBeTruthy();
		expect(component.find('img').prop('alt')).toMatch(name);
		expect(component.find('img').prop('srcSet')).toMatch(avatars.small);
	});

	it('should render only with name props', () => {
		const componentOnlyWithName = shallow(<Avatar name={name} />);

		expect(componentOnlyWithName).toBeDefined();

		//To not contain any images
		expect(componentOnlyWithName.find('img').exists()).toBeFalsy();
		expect(componentOnlyWithName.find('.Avatar__image').exists()).toBeFalsy();
	});
});
