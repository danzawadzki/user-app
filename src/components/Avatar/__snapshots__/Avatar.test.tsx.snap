// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component:  should render match snapshot 1`] = `
<div
  className="Avatar "
>
  <img
    alt="Joe Black avatar"
    className="Avatar__image "
    srcSet="https://randomuser.me/api/portraits/thumb/men/65.jpg 480w, https://randomuser.me/api/portraits/med/men/65.jpg 960w, https://randomuser.me/api/portraits/men/65.jpg 1280w"
  />
</div>
`;
